### ISSUE TITLE
Extended.[k8s.io] StatefulSet [k8s.io] Basic StatefulSet functionality [StatefulSetBasic] should allow template updates

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#15401]
```
/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/statefulset.go:284
Jul 21 07:35:48.496: Failed waiting for pods to enter running: timed out waiting for the condition
/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go:303
```
https://ci.openshift.redhat.com/jenkins/view/All/job/merge_pull_request_origin/1333/
@mfojtik @tnozicka @smarterclayton

**Comment by knobunc** [comment]
My failure seems to be due to a PV not binding: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3369/testReport/junit/(root)/Extended/_k8s_io__StatefulSet__k8s_io__Basic_StatefulSet_functionality__StatefulSetBasic__should_allow_template_updates/

**Comment by smarterclayton** [comment]
@gnufied

On Jul 21, 2017, at 9:35 AM, Ben Bennett <notifications@github.com> wrote:

My failure seems to be due to a PV not binding:
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3369/testReport/junit/(root)/Extended/_k8s_io__StatefulSet__k8s_io__Basic_StatefulSet_functionality__StatefulSetBasic__should_allow_template_updates/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/issues/15401#issuecomment-317002313>,
or mute the thread

**Comment by kargakis** [comment]
https://ci.openshift.redhat.com/jenkins/view/All/job/merge_pull_request_origin/1343/

**Comment by bparees** [comment]
seeing this repeatedly in https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1349 which is a merge into 3.6.

**Comment by bparees** [comment]
every merge attempt on https://github.com/openshift/origin/pull/15366 has hit this, it seems permanently broken in the 3.6 branch.

**Comment by kargakis** [comment]
https://ci.openshift.redhat.com/jenkins/view/All/job/merge_pull_request_origin/1362/

**Comment by tnozicka** [comment]
```
Jul 23 11:51:54.033: INFO: At 2017-07-23 11:31:53 -0400 EDT - event for datadir-ss-0: {persistentvolume-controller } ProvisioningFailed: Failed to provision volume with StorageClass "standard": Failed to get AWS Cloud Provider. GetCloudProvider returned &{0xc420b56000 0xc4214fede0 openshift-gce-devel-ci us-central1 us-central1-a [us-central1-f us-central1-b us-central1-c us-central1-a] https://www.googleapis.com/compute/v1/projects/openshift-gce-devel-ci/global/networks/ci-prtest-5a37c28-4661-ocp-network [ci-prtest-5a37c28-4661-ocp] ci-prtest-5a37c28-4661- true 0xc4217d00b0} instead
```

Storage provisioning failed...

**Comment by liggitt** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1348/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3368/
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1364/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3423/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3390/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3432/
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1362/
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/3403/

**Comment by jsafrane** [comment]
> Failed to get AWS Cloud Provider. GetCloudProvider returned ... openshift-gce-devel-ci

Did we switch from AWS to GCE? Did someone update storage classes that are installed there?

**Comment by liggitt** [comment]
that log message is particularly unhelpful... added debug commit in https://github.com/openshift/origin/pull/15428

**Comment by liggitt** [comment]
the failing test is test_pull_request_origin_extended_conformance_gce... I would hope that runs in gce

I see this in the jenkins config:
`
oct provision remote all-in-one --os "rhel" --stage "base" --provider "aws" --discrete-ssh-config --name "${JOB_NAME}_${BUILD_NUMBER}" 
`

Mixing `--provider aws` with GCE seems very odd to me

**Comment by jsafrane** [comment]
I am looking at the test and what objects it creates and it seems to me it uses alpha dynamic provisioning. 

https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/test/e2e/framework/statefulset_utils.go#L720

We removed alpha provisioning in Kubernetes 1.7 and we expect that a default storage class is deployed during installation. I don't know anything about our test infrastructure, where can I find the default storage class yaml? Or how can I deploy some test machine(s) and run the tests on our shared GCE account?

**Comment by tnozicka** [comment]
cc @stevekuznetsov

**Comment by stevekuznetsov** [comment]
The way the job works is a little complicated ... but the `--provider aws` is for the bastion host in the job.
![screenshot from 2017-07-24 06-28-25](https://user-images.githubusercontent.com/7328370/28525638-bb3e169e-7039-11e7-8af6-4a9289c9cf6e.png)

**Comment by stevekuznetsov** [comment]
To be clear the `oct provision ... --provider aws` is correct and has no influence on the rest of the job, including the above error quoted by @jsafrane

**Comment by liggitt** [comment]
wondering if the issue is related to storageclass config defaulting to aws-ebs when no provider is specified - https://github.com/openshift/openshift-ansible/blob/release-3.6/roles/lib_openshift/library/oc_storageclass.py#L1462-L1465

**Comment by jsafrane** [comment]
StorageClass from @stevekuznetsov:

```
- apiVersion: storage.k8s.io/v1
  kind: StorageClass
  metadata:
    annotations:
      storageclass.beta.kubernetes.io/is-default-class: "true"
    creationTimestamp: 2017-07-24T14:13:03Z
    name: standard
    namespace: ""
    resourceVersion: "1200"
    selfLink: /apis/storage.k8s.io/v1/storageclasses/standard
    uid: 34dfd4d3-707a-11e7-87fe-42010a800005
  parameters:
    type: pd-standard
  provisioner: kubernetes.io/aws-ebs
```

`provisioner: kubernetes.io/aws-ebs` is wrong, it should be `gce-pd`. Notice that `type: pd-standard` means that author of the storage class knows it's running on GCE...

PR with a fix: https://github.com/openshift/openshift-ansible/pull/4841

**Comment by kargakis** [comment]
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1371/
https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1370/

**Comment by jsafrane** [comment]
Fixed by openshift/openshift-ansible#4841
